tsfresh/feature_extraction/feature_calculators.py

```diff
@@ -762,11 +766,20 @@ def number_peaks(x, n):
     :return: the value of this feature
```

Comment:
I thought about this feature a little bit. I think there is great potential to speed up calculations here if we also make this an apply one.

Also I have an idea for a speed up: At the moment we compare too many values (`x_reduced > np.roll`). Instead we only have to compare neighbours. We could save only if the neighbour is greater or smaller (-1 if smaller, 1 if bigger). Then we are searching for subsequences. I feel that this way the calculations can be greatly speed up.